sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -427,6 +428,7 @@ class SessionCatalog(
       tableDefinition.copy(identifier = tableIdentifier)
     }
 
+    ResolveDefaultColumns.checkDataSourceSupportsDefaultColumns(tableDefinition)
```

Comment:
Do we also need to handle DataFrame API? For example, I create a table using this:
```
val schema = ...
val df = spark.createDataFrame(schema, ...)
df.write.saveAsTable("...")
```

Reply:
This is not supported currently, the feature is SQL-only as of now. References to columns named "default" will simply return "column not found" errors until then. We can certainly consider adding this feature to the DataFrame API and PySpark later, I would be interested in helping with this.